### PR TITLE
Custom nodes

### PR DESCRIPTION
**Pull request by williamprofit**

### CODE REVIEW

**Comment by lancelotblanchard** [comment]
I've done this PR together with @williamprofit during a LiveShare session. @tomproberts, @gidonkatten it would be great to have your review on it.

File: src/baklava/OptionParams.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export default class OptionParams {
```
Comment (lancelotblanchard) [inline src/baklava/OptionParams.ts]: Especially on this @tomproberts. Basically, I want to be able to communicate from the Custom baklava node to the `TextAreaOption`, but the only way of doing this is bt passing arguments when adding the option. We then need this class to be able to set values from the Custom baklava node and to read them from the TextArea option. Does that look good? We might want to have it more general to fit more purposes.

File: src/nodes/model/custom/Custom.ts [hunk 2]
```diff
@@ -1,47 +1,65 @@
...
   private nodeUpdated(event: any) {
-    if (event.name === 'Number of Inputs') {
-      if (event.option) {
-        if (event.option.value > this.numberOfInputs) {
-          this.addInputs(event.option.value - this.numberOfInputs);
-        } else if (event.option.value < this.numberOfInputs) {
-          this.removeInputs(this.numberOfInputs - event.option.value);
-        }
+    if (event.name === CustomOptions.InlineCode) {
+      const code = event.option.value;
```
Comment (gidonkatten) [inline src/nodes/model/custom/Custom.ts]: Everytime the text changes, we update the code and most probably `parse`. What do you think about adding a debounce so we don't need to always do this? A debounce should save us parsing unnecessary code

File: src/baklava/OptionParams.ts (hunk 1 above)
Comment (gidonkatten) [reply src/baklava/OptionParams.ts]: Baklava already has an `InputOption` which is very similar (just doesn't have label next to the box) so you can look at the way they deal with text input

File: src/nodes/model/custom/Custom.ts (hunk 2 above)
Comment (williamprofit) [reply src/nodes/model/custom/Custom.ts]: it's running smooth for now but could be a good idea, I'll look into it

File: src/baklava/OptionParams.ts (hunk 1 above)
Comment (lancelotblanchard) [reply src/baklava/OptionParams.ts]: Hmm, I looked at it but it still doesn't offer a way of communicating with the Option right? What we want to do here is being able from the Custom baklava node to set a flag `hasError` to true in order to add the class `is-invalid` to the textarea html tag.

File: src/nodes/model/custom/Custom.ts (hunk 2 above)
Comment (lancelotblanchard) [reply src/nodes/model/custom/Custom.ts]: I agree, it could be good!

Comment (tomproberts) [reply src/nodes/model/custom/Custom.ts]: Not to put down your good work, but whenever you type something in the node you get console errors for "badly indented file". I think a debounce would definetly be better, but also could we catch the errors from being printed by the console too?

Comment (williamprofit) [reply src/nodes/model/custom/Custom.ts]: yeah ofc that was for debugging mainly, I'll clean it up

File: src/baklava/OptionParams.ts (hunk 1 above)
Comment (tomproberts) [reply src/baklava/OptionParams.ts]: So what you could do, is set the value of a `TextArea` to be an object, e.g. `{ hasError: boolean, text: string }`. Then whenever you parse the code (and there's an error) in the `nodeUpdated` method of `Custom.ts`, you can simply get the `TextArea` option by doing something like `const option = this.options.get('Inline Code')`, and then with that option you can set the value from the `Custom.ts` node. Something like `option.value = { hasError: true, text: code }` should work, no?

That would be the way I would do it. Personally the `hasError` field makes sense to be part of the option's value, maybe you would disagree though.

Comment (tomproberts) [reply src/baklava/OptionParams.ts]: Then obviously inside the `TextArea.vue` component you can just set the class `is-invalid` by normal JavaScript by checking the field `hasError` of its `value`.

Comment (lancelotblanchard) [reply src/baklava/OptionParams.ts]: That sounds great, I've just implemented this. Feel free to check! 🙂

**Review by tomproberts — APPROVED** [review]
I would much prefer for all the 'badly indented file' errors to be caught rather than just left there, but if it is William's intention to fix it and implement more handling at a later stage, then feel free to merge it now.

**Pull request merged**